### ISSUE TITLE
Submission files and db entries are not deleted after being rejected during upload process

### ISSUE THREAD

**Issue by wopsononock** [pkp/pkp-lib#7828]
If a user uploads a file during the submission process, the file is added here: 
https://github.com/pkp/pkp-lib/blob/33eda9b64ffa952231312b57affed5d79dc1847c/api/v1/submissions/PKPSubmissionFileHandler.inc.php#L283-L286

It can be rejected, triggering an error response, but we don't currently have a way to delete the database content about the file or the file itself during this rejection process.  https://github.com/pkp/pkp-lib/blob/33eda9b64ffa952231312b57affed5d79dc1847c/api/v1/submissions/PKPSubmissionFileHandler.inc.php#L319

Steps to Reproduce:
Using Firefox dev tools
Create a new submission and move to step 2, file upload
Open Dev tools in your browser
Open the Network tab
Upload a test file in the UI
Check the database for the site you're using
SELECT * FROM files order by file_id desc limit 10;
Note the highest incremented fileId, this references the file you just uploaded
Once you see the resulting POST request appear in the Network list in dev tools..
Edit and Resend the request after making the following change to the request body:
Change the value of fileStage to 3 (this corresponds to SubmissionFile::SUBMISSION_FILE_NOTE, which is explicitly blocked in the $notAllowedFileStages array
Resend the request and confirm that you receive a 400 with JSON response containing the error: api.submissionFiles.403.unauthorizedFileStageIdWrite
Recheck the database, noting the new file from the resent (intentionally invalid) request.
Despite the rejection message, we still retain a copy of the file in the file system and a reference to it in the files table

Using OJS 3.3.0.8

**Comment by NateWr** [comment]
The PR to `main` was merged (thanks @wopsononock).

@asmecher I've made a PR against `stable-3_3_0`. My thinking here is that the longer this lingers in a stable version, the large the scale of some future upgrade headache we'll face. Having orphaned files and entries in the `files` table will just contribute to confusion.

PR:
https://github.com/pkp/pkp-lib/pull/7830

Tests:
https://github.com/pkp/ojs/pull/3361
